proxy/src/transport/tls/config.rs

```diff
@@ -236,15 +238,20 @@ impl CommonConfig {
             rustls::ClientConfig::new().get_verifier().verify_server_cert(
                     &root_cert_store,
                     &cert_chain,
-                    settings.service_identity.as_dns_name_ref(),
+                    settings.pod_identity.as_dns_name_ref(),
                     &[]) // No OCSP
                 .map(|_| ())
-                .map_err(Error::EndEntityCertIsNotValid)?;
+                .map_err(|err| {
+                    error!("validating certificate failed for {:?}: {:?}", settings.pod_identity, err);
```

Comment:
NIT: might be better to format the error using `fmt::Display`?

Reply:
> NIT: might be better to format the error using fmt::Display?

The error types intentionally don't implement `Display`.

Reply:
Oh, I thought this was a Rustls error?

Reply:
> Oh, I thought this was a Rustls error?

I changed the formatting of the `Rustls` error to `{}`. However, it formats the underlying error using `{:?}` because the underlying type intentionally doesn't implement `Display`.